### PR TITLE
build-system: fix unminified watch mode.

### PR DESCRIPTION
**Pull request by samouri**
**summary**
Can be reproduced on current master by:
1. `gulp`
2. go to http://localhost:8000/dist/v0/amp-bind-0.1.max.js
3. modify`bind-impl.js` and see explosions.

Screen Shot 2021-03-17 at 6 32 48 PM

### CODE REVIEW

**Review by caroqliu — APPROVED** [review]

File: build-system/tasks/extension-helpers.js [hunk 1]
```diff
@@ -384,8 +386,11 @@ async function doBuildExtension(extensions, extension, options) {
  * @param {?Object} options
  */
 function watchExtension(extDir, name, version, latestVersion, hasCss, options) {
-  const watchFunc = function () {
-    removeFromBabelCache(extDir);
+  const watchFunc = function (modifiedFile) {
+    if (options.minify) {
```
Comment (rsimha) [inline build-system/tasks/extension-helpers.js:390]: Doh! Of course.

File: build-system/tasks/extension-helpers.js [hunk 2]
```diff
@@ -384,8 +386,11 @@ async function doBuildExtension(extensions, extension, options) {
  * @param {?Object} options
  */
 function watchExtension(extDir, name, version, latestVersion, hasCss, options) {
-  const watchFunc = function () {
-    removeFromBabelCache(extDir);
+  const watchFunc = function (modifiedFile) {
```
Comment (rsimha) [inline build-system/tasks/extension-helpers.js:389]: nit: While you're here, shall we standardize all three `watchFunc`s in this PR to use arrow notation?

**Review by rsimha — APPROVED** [review]
Thanks for the fix!

File: build-system/tasks/extension-helpers.js (hunk 2 above)
Comment (samouri) [reply build-system/tasks/extension-helpers.js:387]: I actually think these functions look better as old-style funcs :)

File: build-system/tasks/extension-helpers.js [hunk 3]
```diff
@@ -25,6 +25,9 @@ const {
   verifyExtensionBundles,
   jsBundles,
 } = require('../compile/bundles.config');
+const {
+  removeFromBabelCache: removeFromClosureBabelCache,
```
[the remark this answers is not in the document]
Comment (samouri) [reply build-system/tasks/extension-helpers.js]: Done

**Pull request merged**